design-proposals/network-device-capacity.md

```diff
@@ -0,0 +1,148 @@
+# Network Interface Resource Request
+
+Author: [Edward Haas](edwardh@redhat.com)
+
+Contributors: TBD
+
+State: Draft
+
+## Summary
+Support the ability to define how many network devices a VM may potentially contain.
+Allowing uses to hotplug network interfaces at least up to that desired capacity.
+
+## Motivation
+The introduction of network devices hotplug introduced a limitation
+to the amount of possible devices one can attach.
+
+Users of libvirt who can estimate how many potential network devices are needed,
+can define the additional PCI/e controllers in advance.
+KubeVirt does not expose such domain details through its API.
+But even if these details had been exposed by the VMI object,
+a common user would find it challenging to define such low domain details.
+
+### Goals
+- Allow users to define how many network devices can be potentially hotplug.
```

Comment:
For clarity, I would prefer if you made sure the words minimum number of hot-plug gable network devices are featured here. (or something similar ...)

Reply:
I used "guaranteed", hope this is better.